astropy/units/tests/test_quantity_array_methods.py

```diff
@@ -168,6 +168,10 @@ def test_mean_inplace(self):
         assert qi2 is qi
         assert qi == 3.6 * u.m
 
+    def test_mean_where(self):
+        q1 = np.array([1., 2., 4., 5., 6., 7.]) * u.m
+        assert_array_equal(np.mean(q1, where=q1 < 7), 3.6 * u.m)
```

Comment:
Feel free to ignore, but the test assertion is immediately obvious if it's Gaussian summation.

```suggestion
        q1 = np.array([1., 2., 3., 4., 5., 6., 7.]) * u.m
        assert_array_equal(np.mean(q1, where=q1 < 7), 3.5 * u.m)
```

Reply:
I agree that is more clear.

The sequence I used was used in the two previous tests concerning `np.mean()`, `test_mean()` and `test_mean_inplace()`, and I thought I would keep things consistent. What do you think is most important?

Reply:
For me, clarity. I'm even tempted to change the other tests for consistency 😆

Reply:
Do you think changing the other tests is within the scope of this PR?

Reply:
I think best to leave it.